lib/sinon/util/fake_xml_http_request.js

```diff
@@ -545,7 +545,7 @@ extend(FakeXMLHttpRequest.prototype, sinonEvent.EventTarget, {
         this.requestHeaders = {};
         this.responseHeaders = {};
 
-        if (this.readyState > FakeXMLHttpRequest.UNSENT && this.sendFlag) {
+        if (this.readyState > FakeXMLHttpRequest.UNSENT && this.sendFlag && this.readyState < FakeXMLHttpRequest.DONE) {
```

Comment:
It's more explicit in my opinion. Do you care that you're in the done state or do you care that you're in a state that is less than the done state?